devices/Ds1302/Ds1302.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device;
+using System.Device.Gpio;
+using System.Device.Model;
+using Iot.Device.Common;
+
+namespace Iot.Device.Rtc
+{
+    /// <summary>
+    /// Realtime Clock Ds1302.
+    /// </summary>
+    [Interface("Realtime Clock Ds1302")]
+    public class Ds1302 : RtcBase, IDisposable
+    {
+        private readonly GpioController _controller;
+        private readonly GpioPin _clockPin;
+        private readonly GpioPin _resetPin;
+        private readonly GpioPin _dataPin;
+        private readonly byte enableWrite = 0b1000_0000;
+        private readonly byte disableWrite = 0b0000_0000;
+        private readonly byte enableWriteOrRead = 0b1000_0001;
+        private PinMode _currentDataPinDirection;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Ds1302"/> class.
+        /// </summary>
+        /// <param name="controller">Gpio controller.</param>
+        /// <param name="clockPin">Clock pin number.</param>
+        /// <param name="dataPin">Data pin number.</param>
+        /// <param name="resetPin">Reset pin number.</param>
+        /// <exception cref="ArgumentNullException">When controller was not provided.</exception>
+        public Ds1302(GpioController controller, int clockPin, int dataPin, int resetPin)
+        {
+            _controller = controller ?? throw new ArgumentNullException();
+            _clockPin = _controller.OpenPin(clockPin, PinMode.Output);
```

Comment:
you should check that all the pins are valid >=0 (except resetPin, see a bit after). I know, an exception will be thrown by the native part but anyway, best to check here.